include/dxc/Test/HlslTestUtils.h

```diff
@@ -604,12 +631,23 @@ inline bool CompareFloatRelativeEpsilon(
 
 inline bool CompareHalfULP(const uint16_t &fsrc, const uint16_t &fref,
                            float ULPTolerance) {
+  // Treat +0 and -0 as equal
+  if ((fsrc & ~FLOAT16_BIT_SIGN) == 0 && (fref & ~FLOAT16_BIT_SIGN) == 0)
+    return true;
   if (fsrc == fref)
     return true;
   if (isnanFloat16(fsrc))
     return isnanFloat16(fref);
```

Comment:
I actually think the following would be more robust/correct for this case:
```suggestion
  bool nanRef = isnanFloat16(fref);
  bool nanSrc = isnanFloat16(fsrc);
  if (nanSrc || nanRef)
      return (nanSrc && nanRef);
```

Basically, we should have a reference value (fref) of NaN if we expect NaN, and if so, only NaN is valid for fsrc, otherwise, fsrc should not be NaN.

Without this change, it's possible we have fref=NaN, but fsrc is zero or a denorm close to zero, and just because of the particular NaN's bit pattern and the monotonic ordering, we end up with an adjacent value for the ULP comparison, which would be an unintended pass.